corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -537,7 +541,7 @@ def save_new_form(form):
             operation.form = form
 
         try:
-            with transaction.atomic(using=db_name):
+            with transaction.atomic(using=db_name, savepoint=False):
```

Comment:
We don't need savepoints since if anything fails we want the whole transaction to fail.

Reply:
@snopoke I'm trying to understand why `savepoint=False` was added here rather than removing `with transaction.atomic(...)`. Do you remember?

Context: this function has a `@transaction.atomic` decorator (with the default `savepoint=True`), so it seems this inner transaction block is not necessary. Also `savepoint=False` has this somewhat confusing caveat:

> This option should only be used if the overhead of savepoints is noticeable. It has the drawback of breaking the error handling described above. [[source](https://docs.djangoproject.com/en/1.11/topics/db/transactions/#django.db.transaction.atomic)]

Reply:
I think the decorator on the function shouldn't be there (it's not even on the write database).

I think the atomic context here is because this function can be called by itself (in the case of error handling) or as part of normal form processing (in which case there is a parent transaction). Fine to move the atomicity out of the function and handle it differently in the error case.